flow/connectors/postgres/qrep_query_executor.go

```diff
@@ -409,7 +412,25 @@ func (qe *QRepQueryExecutor) mapRowToQRecord(
 
 	record := make([]types.QValue, len(fds))
 	for i, fd := range fds {
-		tmp, err := qe.parseFieldFromPostgresOID(fd.DataTypeOID, fd.TypeModifier, values[i], qe.customTypeMapping, qe.version)
+		var nullable bool
+		if i < len(schema.Fields) && schema.Fields[i].Name == fd.Name {
+			nullable = schema.Fields[i].Nullable
+		} else {
+			for _, field := range schema.Fields {
+				if field.Name == fd.Name {
+					nullable = field.Nullable
+					break
+				}
```

Comment:
looks like mapRowToRecord is in the hot path, could/should we aim to do some preprocessing before `for rows.Next() {...}` loop is entered to make this O(1)